### PR TITLE
Added local configuration for Hugging Face cache directory

### PR DESCRIPTION
**Pull request by wangsrGit119**
## Changes
Added local configuration for Hugging Face cache directory

### Details
- Set `HF_HOME` environment variable to `hf_home` subdirectory under current working directory
- Automatically create the directory if it doesn't exist
- Print `HF_HOME` path for debugging purposes

### Rationale
- Avoid Hugging Face models and cache files being saved in user's home directory by default
- Keep project dependencies and cache files contained within project directory for better management
- Helps maintain consistent file structure across different environments

### Impact
- All Hugging Face related downloads and cache will be stored in project's `hf_home` directory
- Does not affect Hugging Face configurations for other system projects

### CODE REVIEW

**Comment by Byaidu** [comment]
its strange to store model files everywhere instead of upstream default location

**Comment by wangsrGit119** [comment]
Perhaps most people prefer to have control over where exactly the project dependency model is stored.

File: pdf2zh/pdf2zh.py
Comment (Wybxc) [inline pdf2zh/pdf2zh.py:1]: This modification forces all models to be downloaded into the current folder. However, this approach has a few issues:  
1. Users might run the script in different folders, making it impossible to locate the downloaded models.  
2. If users prefer to use the default Hugging Face download directory, they should have the freedom to do so.

File: Dockerfile.zh
Comment (Wybxc) [inline Dockerfile.zh:1]: This is a file unrelated to this contribution; please submit it through a separate Pull Request.

**Review by Wybxc — CHANGES_REQUESTED** [review]

File: pdf2zh/pdf2zh.py
Comment (wangsrGit119) [reply pdf2zh/pdf2zh.py:1]: download completed and clean ?

File: Dockerfile.zh
Comment (Wybxc) [reply Dockerfile.zh:1]: GitHub's pull requests are organized around branches. You need to use two different branches to submit PRs separately.
